catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +251,109 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+                    .is()
+                    .equalTo()
+                    .text(source);
+            Query query = new QueryImpl(filter);
+
+            QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(query));
+
+            List<Result> metacards = queryResponse.getResults();
+            if (metacards != null && metacards.size() > 0) {
+                Metacard metacard = metacards.get(0)
+                        .getMetacard();
+                if (metacard != null) {
+                    List<Serializable> currentlyPublishedLocations = metacard.getAttribute(
+                            RegistryObjectMetacardType.PUBLISHED_LOCATIONS)
+                            .getValues();
+
+                    List<String> unpublishLocations = new ArrayList<>();
+
+                    unpublishLocations = currentlyPublishedLocations.stream()
+                            .filter(location -> !destinations.contains((String) location))
+                            .map(location -> (String) location)
+                            .collect(Collectors.toList());
+
+                    for (String id : destinations) {
+                        CreateRequest createRequest = new CreateRequestImpl(metacard);
+                        try {
+                            CreateResponse createResponse = catalogStoreMap.get(id)
+                                    .create(createRequest);
+                            if (createResponse.getProcessingErrors()
+                                    .isEmpty()) {
+                                updatedPublishedLocations.addAll(createResponse.getCreatedMetacards()
+                                        .stream()
+                                        .filter(createdMetacard -> createdMetacard.getAttribute(
+                                                RegistryObjectMetacardType.REGISTRY_ID)
+                                                .equals(metacard.getAttribute(
+                                                        RegistryObjectMetacardType.REGISTRY_ID)))
+                                        .map(createdMetacard -> id)
+                                        .collect(Collectors.toList()));
+                            }
+                        } catch (IngestException e) {
+                            LOGGER.error("Unable to create metacard in catalogStore %s", id, e);
+                            Filter createFilter =
+                                    filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                            .is()
+                                            .equalTo()
+                                            .text(source);
+                            Query createQuery = new QueryImpl(filter);
+                            SourceResponse createQueryResponse = catalogStoreMap.get(id)
+                                    .query(new QueryRequestImpl(createQuery));
```

Comment:
Might create a helper method that does the filter building and query since we do it in two locations. It could be something like checkIfMetacardExists(String location) that returns a boolean